cpp/src/neighbors/detail/cagra/cagra_search.cuh

```diff
@@ -41,8 +41,187 @@
 
 #include <rmm/cuda_stream_view.hpp>
 
+#include <cub/cub.cuh>
+#include <thrust/copy.h>
+#include <thrust/sequence.h>
+
 namespace cuvs::neighbors::cagra::detail {
 
+template <typename T, typename IdxT>
+void merge_top_k_gpu(raft::resources const& res,
```

Comment:
We have routines to do this top-k merge already. We should use the same centralized routines so that we aren't reinventing this. I know the multi-gpu (sharded) search implementation should be using the same centralized routines also. cc @benfred @viclafargue

Reply:
Can I have the code link for the existing tok-k merging?

Reply:
Hi @cjnolet @benfred @viclafargue , I suppose the existing tok-k merge API that Corey mentioned here is the `knn_merge_parts`, If that's correct, the `knn_merge_parts` seems to be different from the `merge_top_k_gpu `. Anyway, I will move the `merge_top_k_gpu ` to a centralized location.

Reply:
Done